extensions/mqtt/processors/PublishMQTT.cpp

```diff
@@ -76,15 +67,15 @@ void PublishMQTT::onSchedule(core::ProcessContext *context, core::ProcessSession
 }
 
 void PublishMQTT::onTrigger(const std::shared_ptr<core::ProcessContext> &context, const std::shared_ptr<core::ProcessSession> &session) {
-  std::shared_ptr<core::FlowFile> flowFile = session->get();
-
-  if (!flowFile) {
+  if (!reconnect()) {
```

Comment:
This way we reconnect even if there is no FlowFile to publish. I am not saying that it is necessarily a problem, but it changes the previous behaviour. Is this intentional?

Reply:
Reconnect doesn't do anything unless the connection is interrupted. 
If we have an existing connection (expected case), it just returns true.

Reply:
Yeah, that's the conclusion I reached as well, should be fine, thanks.